src/DDTrace/Tracer.php

```diff
@@ -374,7 +383,13 @@ public function getTracesAsArray()
                 // the internal (hard-coded) processors programmatically.
 
                 $this->traceAnalyticsProcessor->process($span);
-                $traceToBeSent[] = SpanEncoder::encode($span);
+                $encodedSpan = SpanEncoder::encode($span);
+                if ($sendTracesViaThread) {
+                    $messagePack = dd_trace_serialize_msgpack($encodedSpan);
+                    dd_trace_coms_flush_span($groupId, $messagePack);
```

Comment:
`ddtrace_coms_next_group_id()` is used to group spans related to the same trace. Normally a trace_id could be used. But its a 63 bit integer and who knows if it will not turn into a full on string in the future. 
We could support it a bit more on C level but either solution that comes to my mind will be much more complex than the existing approach.  I think its a great idea, to make it more transparent to the PHP, but if possible I'd suggest we tackle this in future Pull requests since this one is already quite complex.

Reply:
as for combining dd_trace_coms_flush_span into one call - This should be very doable - let me try it out

Reply:
Done